### ISSUE TITLE
Setting Significant Digits on NSNumberFormatter causes problems when changing values

### ISSUE THREAD

**Issue by Vortec4800** [FormatterKit/FormatterKit#155]
I've run into some issues with the provided NSNumberFormatter when configuring the number of digits.

The configuration I want is at least 1 integer digit, and 1 fraction digit. So 1.5 miles, 10.2 miles, and 300.0 miles for example.

The code in TTTLocationFormatter to create the NSNumberFormatter is as follows:

```
_numberFormatter = [[NSNumberFormatter alloc] init];
[_numberFormatter setLocale:[NSLocale currentLocale]];
[_numberFormatter setNumberStyle:NSNumberFormatterDecimalStyle];
[_numberFormatter setMaximumSignificantDigits:2];
[_numberFormatter setUsesSignificantDigits:YES];
```

Configuring the formatter in my own code is as follows:

```
TTTLocationFormatter *formatter = [[TTTLocationFormatter alloc] init];
[formatter setBearingStyle:TTTBearingAbbreviationWordStyle];
formatter.numberFormatter.usesSignificantDigits = NO;
formatter.numberFormatter.minimumIntegerDigits = 1;
formatter.numberFormatter.minimumFractionDigits = 1;
formatter.numberFormatter.maximumFractionDigits = 1;
```

This does not provide the correct results, even though I'm disabling the uses significant digits property. The output is the same as if I didn't do any configuration, which is 8.2 miles, 10 miles, 100 miles, etc.

If I remove the last two lines in the TTT init so it looks like this:

```
_numberFormatter = [[NSNumberFormatter alloc] init];
[_numberFormatter setLocale:[NSLocale currentLocale]];
[_numberFormatter setNumberStyle:NSNumberFormatterDecimalStyle];
```

everything works as expected.

I believe this is actually an Apple bug, but I wanted to bring it up here. Perhaps the TTT default should be different, and not use significant digits as they seem to be broken?

**Comment by Vortec4800** [comment]
As a side note, I've also created a Radar for this issue at radar://19953915.

**Comment by vanderneut** [comment]
I have ran into this exact same problem. I want to display dollar amounts with 3 significant digits and 2 fraction digits, e.g. converting "$19.17" to "$19.20". However, I keep getting back "$19.2" even though I have set my min and max fraction digits to 2. Frustrating :-(

**Comment by harlanhaskins** [comment]
I'm having this same issue here.

**Comment by mattt** [comment]
Apologies for taking so long to respond, @Vortec4800. This is now fixed by 2bde467, which removes those two problematic lines you mentioned.
